organization/src/integrationTest/kotlin/com/cosmotech/organization/service/OrganizationServiceIntegrationTest.kt

```diff
@@ -1283,1447 +1283,1502 @@ class OrganizationServiceIntegrationTest : CsmRedisTestBase() {
+                createTestOrganizationWithSimpleSecurity(
+                    name, TEST_ADMIN_USER_ID, ROLE_NONE, ROLE_NONE))
+        runAsPlatformAdmin()
         assertThrows<CsmResourceNotFoundException> {
-          val name = "o-connector-test-1"
-          val organizationRegistered =
-              organizationApiService.registerOrganization(createTestOrganization(name))
-
-          organizationApiService.updateOrganizationAccessControl(
-              organizationRegistered.id!!, UNKNOWN_IDENTIFIER, OrganizationRole(role = ROLE_EDITOR))
+          organizationApiService.getOrganizationAccessControl(
+              organizationRegistered.id!!, TEST_USER_ID)
         }
-      }
-
-      @Test
-      fun `updateOrganizationAccessControl with owned organization and wrong role`() {
-        assertThrows<CsmClientException> {
-          val name = "o-connector-test-1"
-          val organizationRegistered =
-              organizationApiService.registerOrganization(createTestOrganization(name))
+        val otherUserACL = OrganizationAccessControl(id = TEST_USER_ID, role = ROLE_VIEWER)
+        assertNotNull(
+            organizationApiService.addOrganizationAccessControl(
+                organizationRegistered.id!!, otherUserACL))
 
-          organizationApiService.updateOrganizationAccessControl(
-              organizationRegistered.id!!,
-              TEST_USER_ID,
-              OrganizationRole(role = UNKNOWN_IDENTIFIER))
-        }
+        val otherUserACLRetrieved =
+            organizationApiService.getOrganizationAccessControl(
+                organizationRegistered.id!!, TEST_USER_ID)
+        assertEquals(otherUserACL, otherUserACLRetrieved)
       }
+    }
 
-      @Test
-      fun `updateOrganizationAccessControl with not owned organization, WRITE_SECURITY permission, can update user (!= ROLE_NONE)`() {
-        assertDoesNotThrow {
-          val name = "o-connector-test-1"
-          runAsDifferentOrganizationUser()
-          val organizationRegistered =
-              organizationApiService.registerOrganization(createTestOrganization(name))
+    @Test
+    fun `updateOrganizationAccessControl with owned organization cannot update last admin`() {
+      assertThrows<CsmAccessForbiddenException> {
+        val name = "o-connector-test-1"
+        val organizationRegistered =
+            organizationApiService.registerOrganization(createTestOrganization(name))
+        organizationApiService.updateOrganizationAccessControl(
+            organizationRegistered.id!!, TEST_ADMIN_USER_ID, OrganizationRole(role = ROLE_VIEWER))
+      }
+    }
 
-          organizationApiService.addOrganizationAccessControl(
-              organizationRegistered.id!!,
-              OrganizationAccessControl(id = TEST_ADMIN_USER_ID, role = ROLE_ADMIN))
+    @Test
+    fun `updateOrganizationAccessControl with owned organization can update user (!= ROLE_NONE)`() {
+      assertDoesNotThrow {
```

Comment:
assertDoesNotThrow should be specific else it's the same as not having it

Reply:
same comment as above